nautobot_circuit_maintenance/tests/test_sources.py

```diff
@@ -313,40 +328,70 @@ class TestGmailAPISource(TestCase):
 
     def setUp(self):
         """Prepare data for tests."""
-        settings.PLUGINS_CONFIG = {"nautobot_circuit_maintenance": {"notification_sources": [SOURCE_GMAIL_API.copy()]}}
+        settings.PLUGINS_CONFIG = {
+            "nautobot_circuit_maintenance": {
+                "notification_sources": [SOURCE_GMAIL_API_SERVICE_ACCOUNT.copy(), SOURCE_GMAIL_API_OAUTH.copy()]
+            }
+        }
         # Deleting other NotificationSource and Provider to define a reliable state.
-        NotificationSource.objects.exclude(name__in=[SOURCE_GMAIL_API["name"]]).delete()
-        self.source = NotificationSource.objects.get(name=SOURCE_GMAIL_API["name"])
+        NotificationSource.objects.exclude(name__in=[SOURCE_GMAIL_API_SERVICE_ACCOUNT["name"]]).delete()
+        self.source = NotificationSource.objects.get(name=SOURCE_GMAIL_API_SERVICE_ACCOUNT["name"])
 
-    def test_source_factory(self):
+        with open(SOURCE_GMAIL_API_SERVICE_ACCOUNT["credentials_file"], "w") as credentials_file:
+            json.dump({"type": "service_account"}, credentials_file)
+
+        with open(SOURCE_GMAIL_API_OAUTH["credentials_file"], "w") as credentials_file:
+            json.dump({}, credentials_file)
```

Comment:
Can we provide more "realistic" JSON data than these minimal dicts? What other keys are typically present in these files in reality?

Reply:
I could add more data, but I tried to keep it minimal to the actual code validation in factory method `init` (now updated with the oauth check). 
What would you like to test more regarding them?

Reply:
Not sure. I guess I was assuming that the calls out to the Google libraries would need to do some sort of validation of this data, but perhaps not.

Reply:
I just defined them to go over the factory pattern and get the proper class, I'm not doing any kind of integration tests with Google API at the moment.